File: II Anno/POO/Laboratorio/Pasquale/12102023/Triangolo.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Inserisci base");
        double base = scan.nextDouble();
        System.out.println("Inserisci altezza");
        double altezza = scan.nextDouble();

        Triangolo myTriangle = new Triangolo(base, altezza);
        System.out.println("L'area e': " + myTriangle.calcolaArea());
    }
}

class Triangolo {
    double base;
    double altezza;

    public Triangolo(double base, double altezza) {
        System.out.println("Creo oggetto triangolo");
        this.base = base;
        this.altezza = altezza;
    }

    double calcolaArea() {
        return (base * altezza) / 2;
    }
}

class Ciao {
    static int cont;

    public Ciao() {
        
    }
}

File: II_Anno/POO/Laboratorio/Pasquale/12102023/Triangolo.cpp
#include <iostream>
using namespace std;

class Triangle {
public:
    double base;
    double altezza;

    Triangle(double base, double altezza) {
        this->base = base;
        this->altezza = altezza;
    }

    double calcolaArea() {
        return (base * altezza) / 2;
    }
};

int main() {
    double base, altezza;
    cout << "Inserisci base" << endl;
    cin >> base;
    cout << "Inserisci altezza" << endl;
    cin >> altezza;

    Triangle *my_triangle = new Triangle(base, altezza);
    cout << "L'area e': " << my_triangle->calcolaArea() << endl;

    return 0; 
}